kube-runtime/src/controller/mod.rs

```diff
@@ -319,6 +327,13 @@ where
                     None => future::err(Error::ObjectNotFound(request.obj_ref.erase())).right_future(),
                 }
             })
+            .delay_tasks_until(async move {
+                tracing::debug!("applier runner held until store is ready");
+                let res = delay_store.wait_until_ready().await;
+                tracing::debug!("store is ready, starting runner");
+                res
```

Comment:
While it looks like there's a lot of complexity down in the runner (that i didn't get to cover fully in the first pass here), I am very positive that you've managed to do contain this horrible polling dance somewhere else, without making the complexity of this one critical function worse.